app/javascript/redux/api.js

```diff
@@ -0,0 +1,10 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+
+const fetchData = createAsyncThunk('GREETING/FETCHRANDOM', async () => {
+  const response = await fetch('http://localhost:3000/root/greetings');
```

Comment:
- [ ] Also, since both apps come from the same origin, I suggest you only use the endpoint, like this:

```suggestion
  const response = await fetch('/root/greetings');
```

This is because I'm currently getting a CORS error:

![image](https://github.com/Elerqsousy/react-on-rails-one-app/assets/77069011/6dc29e7b-1e32-446c-8ad3-2b4327c473f9)